### ISSUE TITLE
Scheduler broken in 'develop': RuntimeError: dictionary changed size during iteration

### ISSUE THREAD

**Issue by whiteinge** [saltstack/salt#22339]
Bug is not present on 2015.2.

```python
[INFO    ] Running scheduled job: test_master_side_syntax
[DEBUG   ] schedule: This job was scheduled with jid_include, adding to cache (jid_include defaults to True)
[DEBUG   ] schedule: This job was scheduled with a max number of 1
[DEBUG   ] schedule.handle_func: adding this job to the jobcache with data {'fun': 'cache.grains', 'jid': '20150403130744828718', 'pid': 6946, 'id': 'Burke.local_master', 'schedule': 'test_master_side_syntax'}
[ERROR   ] Unhandled exception running cache.grains
Traceback (most recent call last):
  File "/Users/shouse/src/salt/salt/salt/utils/schedule.py", line 558, in handle_func
    ret['return'] = self.functions[func](*args, **kwargs)
  File "/Users/shouse/src/salt/salt/salt/client/mixins.py", line 76, in wrapper
    for kwargs_key, kwargs_value in six.iteritems(kwargs):
RuntimeError: dictionary changed size during iteration
```

**Comment by rallytime** [comment]
ping @garethgreenaway

**Comment by garethgreenaway** [comment]
@rallytime saw this the other day. Hope to have a fix submitted today.

**Comment by rallytime** [comment]
Awesome! Thanks @garethgreenaway! :)

**Comment by garethgreenaway** [comment]
Not exactly the thing I saw the other day but I think I've identified the issue.  PR coming shortly.

**Comment by garethgreenaway** [comment]
See PR, this didn't happen in 2015.2 since it's using kwargs.items() where develop is using six.iteritems().

**Comment by cachedout** [comment]
Merged #22351 so I'll go ahead and close this. Thanks all!
